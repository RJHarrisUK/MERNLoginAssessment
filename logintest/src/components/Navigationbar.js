import React from 'react';
import { Link } from "react-router-dom";
import {
    Navbar,
    NavbarBrand,
    NavbarToggler,
    Collapse,
    Nav,
    NavItem,
    NavLink
} from 'reactstrap';

export default class Navigationbar extends React.Component {

    constructor(props) {
        super(props);

        this.toggle = this.toggle.bind(this);
        this.state = {
            isOpen: false
        };
    }
    toggle() {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }
    render() {
        return (
            <div>
                <Navbar color="dark" dark expand="md">
                    <NavbarBrand href="/">Register User</NavbarBrand>
                    <NavbarToggler onClick={this.toggle} />
                    <Collapse isOpen={this.state.isOpen} navbar>
                        <Nav className="ml-auto" navbar>
                            <NavItem>
                                <NavLink href="/Login">Login User</NavLink>
                            </NavItem>
                            <NavItem>
                                <NavLink href="/List">Show Users</NavLink>
                            </NavItem>
                        </Nav>
                    </Collapse>
                </Navbar>
            </div >
        )
    }
}
