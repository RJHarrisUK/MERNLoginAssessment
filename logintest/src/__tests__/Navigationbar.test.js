import React from "react";
import { mount } from "enzyme";
import Navigationbar from '../components/Navigationbar.js';

it("render a navigation bar comp", () => {

    const wrapper = mount(<Navigationbar />);

    expect(wrapper).toMatchSnapshot();

}); 