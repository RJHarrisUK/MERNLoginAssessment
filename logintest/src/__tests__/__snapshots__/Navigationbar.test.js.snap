// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a navigation bar comp 1`] = `ReactWrapper {}`;

exports[`render a user 1`] = `ReactWrapper {}`;
